### PR TITLE
fix: Graph2 centerOnItem works as expected

### LINKED ISSUE #3651: Graph2 centerOnItem not working as expected

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #3651

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/graph/src/graph2/graph.ts [hunk 1]
```diff
@@ -352,7 +352,12 @@ export class Graph2 extends SVGZoomWidget {
             [x, y] = [0, 0]; // center(item.points);
         }
         if (x !== undefined && y !== undefined) {
-            const bbox = item.element.node().getBBox();
+            const node = item.element.node();
+            const bbox = node.getBBox();
+            const xforms = node.getAttribute('transform');
+            const parts  = /translate\(\s*([^\s,)]+)[ ,]([^\s,)]+)/.exec(xforms);
+            x = +parts[1];
+            y = +parts[2];
```
Comment (GordonSmith) [inline packages/graph/src/graph2/graph.ts:360]: I suspect the "project", "projectPlaceholder" and "rproject" methods would be better here (they ultimately are what created the translate that your parsing?

File: packages/graph/src/graph2/graph.ts [hunk 2]
```diff
@@ -353,6 +353,9 @@ export class Graph2 extends SVGZoomWidget {
         }
         if (x !== undefined && y !== undefined) {
             const bbox = item.element.node().getBBox();
+            const proj = this.projectPlacholder(item as VertexPlaceholder);
```
Comment (GordonSmith) [inline packages/graph/src/graph2/graph.ts:356]: Forcing to a VertexPlaceholder is dangerous, put it in a `this._graphData.isVertex(item)` and you can remove the "as"

Comment (GordonSmith) [reply packages/graph/src/graph2/graph.ts:356]: (item might also be a SubgraphPlaceholder or an EdgePlaceholder) so you may need to rework the logic a little bit...

**Comment by jbrundage** [comment]
@GordonSmith this is resolved and working as expected

**Comment by GordonSmith** [comment]
Did you test "centerOnItem" with an edge and subgraph, or just the vertex?

**Comment by jbrundage** [comment]
@GordonSmith yes I tested it with edge and subgraph as well

File: packages/graph/src/graph2/graph.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/graph/src/graph2/graph.ts:356]: I suspect this code should be relocated to line 348->350

**Pull request merged**